tools/input-segments/src/InputSegments.ts

```diff
@@ -415,163 +508,416 @@ export class InputSegments extends TextfieldBase {
-        let day = this.daySegment?.value;
+        // We always use the first day of a month unless a specific day is specified
+        let day = this.daySegment?.value ?? 1;
 
-        // When only date segments are being used
-        if (this.includeDate && !this.includeTime) {
-            if (isNumber(year) && isNumber(month) && isNumber(day)) {
-                this.newDateTime = new CalendarDateTime(year, month, day);
-            }
+        // We always use the first month of the year unless a specific month is specified
+        let month =
+            this.monthSegment?.value ??
+            getMinimumMonthInYear(this.currentDateTime);
 
+        let year = this.yearSegment?.value ?? this.currentDateTime.year;
+        let hour = this.hourSegment?.value ?? this.currentDateTime.hour;
+        let minute = this.minuteSegment?.value ?? this.currentDateTime.minute;
+        let second = this.secondSegment?.value ?? this.currentDateTime.second;
+
+        switch (segment.type) {
+            case 'day':
+                day = segment.value;
+                break;
+            case 'month':
+                month = segment.value;
+                break;
+            case 'year':
+                year = segment.value;
+                break;
+            case 'hour':
+                hour = segment.value;
+                break;
+            case 'minute':
+                minute = segment.value;
+                break;
+            case 'second':
+                second = segment.value;
+                break;
+            case 'dayPeriod':
+                hour = (segment.value ?? 0) + 1;
+                break;
+        }
+
+        /**
+         * For the year we do not use the value returned by the formatter, to avoid that the typed year is displayed in
+         * an unexpected way. For example, when typing “2”, the year would be formatted as “1902”, but we keep it as it
+         * is being displayed on the screen. If the user wants to enter the year “1902”, he will enter number by number
+         */
+        if (segment.type === 'year') {
+            segment.formatted = String(year);
             return;
         }
 
-        // When only time segments are being used, we need to set the date based on the current date
-        if (!this.includeDate) {
-            year = this.currentDateTime.year;
-            month = this.currentDateTime.month;
-            day = this.currentDateTime.day;
+        /**
+         * If the day being formatted is February 29th but the year segment has not yet been filled, we need to use a
+         * leap year to allow the 29th to remain, otherwise, if we use the current year and it is not a leap year, the
+         * day that would be displayed would be March 1st, as February 29th would not exist and JavaScript “moves” the
+         * day to the next day. As this year is only used to format the day and month, we use the year 2000 as the "base
+         * year" for formatting
+         */
```

Comment:
🤯 How did you catch this!? Great work. Hopefully this is in the tests either in actuality or as a stub.

Reply:
Is there a test or stub for this?